test/extended/networking/egress_firewall.go

```diff
@@ -37,18 +39,38 @@ var _ = g.Describe("[sig-network][Feature:EgressFirewall]", func() {
 	InOVNKubernetesContext(
 		func() {
 			g.It("should ensure egressfirewall is created", func() {
-				doEgressFwTest(egFwf, egFwoc, oVNKManifest, true)
+				doEgressFwTest(egFwf, egFwoc, oVNKManifest, true, false)
 			})
 		},
 	)
 	// For Openshift SDN its supports EgressNetworkPolicy objects
 	InOpenShiftSDNContext(
 		func() {
 			g.It("should ensure egressnetworkpolicy is created [apigroup:network.openshift.io]", func() {
-				doEgressFwTest(egFwf, egFwoc, openShiftSDNManifest, false)
+				doEgressFwTest(egFwf, egFwoc, openShiftSDNManifest, false, false)
 			})
 		},
 	)
+
+	// When OVNKubernetes subnet and coredns-ocp-dnsnameresolver plugins are enabled.
```

Comment:
Instead of checking whether the cluster is techpreview you should create featuregate specific tests:
https://github.com/openshift/api?tab=readme-ov-file#defining-featuregate-e2e-tests
Example: https://github.com/openshift/origin/pull/28719

Reply:
I am thinking we will make a whole new context in the next dev cycle following all instructions ^, but we wanted 1 simple test in the existing context before we even merge tech preview feature. Hopefully that makes sense

Reply:
not sure if this helps, but some existing tests actually were an example for this one https://github.com/openshift/origin/blob/75f2c029adf1f56413a95bf0118052a00db41c6a/test/extended/builds/volumes.go#L161-L163

Reply:
The approach of testing featuregates is fairly new as far as I know. I think it is better than checking for techpreview because it doesn't require any followup work. It should be really easy to test whether it works in the existing `g.It`

Reply:
suggestion incorporated. New `Describe` block added with `OCPFeatureGate` tag.